app/views/surveys/_educational_fields.html.erb

```diff
@@ -0,0 +1,165 @@
+<% if survey.id.nil? %>
```

Comment:
If `id` is `nil` I assume the `survey` variable can be empty as well. If that's the case it might be worth adding a check on the variable itself as well.

Reply:
@Levia the reason for this is basically on the new survey the id is nil but the survey does exist, but when the survey is edited the id is not nil and again the survey does exist. This is the only thing I am checking against here. Perhaps there is a better way? Or perhaps I should comment why I have done this in the code?

Reply:
Uhm, usually forms should work regardless and fetch the data if it's existing or the fields beign empty if not.
This looks like a normal form to me. Perhaps the only thing is that you need to make sure that the object is initialised or fetched correctly in the controller.

For example, for a new survey there should be something like: `@survey = Survey.new`.
For an existing one, it should be `@survey = Survey.find(params[:id])`.
So you can use the `@survey` object regardless

Reply:
Ah sure thanks @Levia, the thing is that I am showing certain fields as editable on the create but only two are editable on the editing of the survey. This is because they cannot change those fields after they create the survey.